src/khoj/database/adapters/__init__.py

```diff
@@ -520,6 +527,7 @@ def save_conversation(
             conversation = Conversation.objects.filter(user=user, client=client_application, id=conversation_id)
         else:
             conversation = Conversation.objects.filter(user=user, client=client_application)
+
         if conversation.exists():
             conversation.update(conversation_log=conversation_log, slug=slug, updated_at=datetime.now(tz=timezone.utc))
```

Comment:
What happens if the chat API is called with no `conversation_id` or `client_application` set?

Reply:
Shouldn't we updated the most recently updated conversation from the retrieved conversation list 
```suggestion
        else:
            conversation = Conversation.objects.filter(user=user, client=client_application)

        if conversation.exists():
            conversation.order_by("-updated-at").first().update(conversation_log=conversation_log, slug=slug, updated_at=datetime.now(tz=timezone.utc))
```

Reply:
Great point! Shouldn't generally happen, but let's get the first conversation last updated.